src/PlusDataCollection/OpenHaptics/vtkPlusOpenHapticsDevice.cxx

```diff
@@ -0,0 +1,294 @@
+/*=Plus=header=begin======================================================
+Program: Plus
+Copyright (c) Laboratory for Percutaneous Surgery. All rights reserved.
+See License.txt for details.
+=========================================================Plus=header=end*/
+
+
+// Local includes
+#include "PlusConfigure.h"
+#include "vtkPlusDataSource.h"
+#include "vtkPlusOpenHapticsDevice.h"
+
+//HD includes
+#include <HDU/hduVector.h>
+#include <HDU/hduError.h>
+#include <HD/hd.h>
+
+// VTK includes
+#include <vtkMatrix4x4.h>
+#include <vtkTransform.h>
+
+// STL includes
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+vtkStandardNewMacro(vtkPlusOpenHapticsDevice);
+
+vtkPlusOpenHapticsDevice::vtkPlusOpenHapticsDevice()
+  :FrameNumber(-1)
+  , DeviceHandle(-1)
+  , DeviceName("PLUS")
+  , toolTransform(vtkSmartPointer<vtkTransform>::New())
+  , rotation(vtkSmartPointer<vtkTransform>::New())
+  , velMatrix(vtkSmartPointer<vtkMatrix4x4>::New())
+  , buttonMatrix(vtkSmartPointer<vtkMatrix4x4>::New())
+  , toolMatrix(vtkSmartPointer<vtkMatrix4x4>::New())
+{
+  this->RequirePortNameInDeviceSetConfiguration = true;
+  this->StartThreadForInternalUpdates = true;
+  this->AcquisitionRate = 20;
+
+  this->rasCorrection = vtkSmartPointer<vtkMatrix4x4>::New();
+  this->rasCorrection->Identity();
+  this->rasCorrection->SetElement(0, 0, -1);
+  this->rasCorrection->SetElement(1, 1, 0);
+  this->rasCorrection->SetElement(1, 2, 1);
+  this->rasCorrection->SetElement(2, 2, 0);
+  this->rasCorrection->SetElement(2, 1, 1);
+}
+
+vtkPlusOpenHapticsDevice::~vtkPlusOpenHapticsDevice(){}
+
+std::string vtkPlusOpenHapticsDevice::GetSdkVersion()
+{
+  std::stringstream ss;
+  ss << HD_VERSION_MAJOR_NUMBER << "." << HD_VERSION_MINOR_NUMBER << "." << HD_VERSION_BUILD_NUMBER;
+  return ss.str();
+}
+
+PlusStatus vtkPlusOpenHapticsDevice::Probe()
+{
+  if (this->Connected)
+  {
+    LOG_ERROR("vtkPlusOpenHapticsDevice::Probe should not be called while the device is already initialized");
+    return PLUS_FAIL;
+  }
+  if (this->InternalConnect())
+  {
+    this->InternalDisconnect();
+    LOG_INFO("vtkPlusOpenHapticsDevice::Probe found OpenHaptics Device");
+    return PLUS_SUCCESS;
+  }
+  else
+  {
+    LOG_ERROR("vtkPlusOpenHapticsDevice::Probe failed to connect to device");
+    return PLUS_FAIL;
+  }
+
+}
+
+PlusStatus vtkPlusOpenHapticsDevice::InternalUpdate()
+{
+  //schedule the internal callback - must be done each time device is polled
+  //the scheduler is intended to manage multiple clients requesting data from the same device
+  hdScheduleSynchronous(positionCallback, this, HD_MAX_SCHEDULER_PRIORITY);
+
+  return PLUS_SUCCESS;
+}
+
+PlusStatus vtkPlusOpenHapticsDevice::InternalConnect()

```

Comment:
Minor code formatting issue: would you add the 
`//----------------------------------------------------------------------------`
separators before each function?

ie:
```
...
}

//----------------------------------------------------------------------------
PlusStatus vtkPlusOpenHapticsDevice::InternalConnect()
```